### PR TITLE
Added v-model binding

### PR DESCRIPTION
**Pull request by ouraios**

### CODE REVIEW

File: Rate.vue [hunk 1]
```diff
@@ -58,6 +58,7 @@ export default {
       this.$emit('beforeRate', this.rate)
       this.rate = index
       this.$emit('afterRate', this.rate)
+      this.$emit('input', this.rate)
```
Comment (SinanMtl) [inline Rate.vue]: You were added 'input' emit but this is already same with 'afterRate'. You can do same things with 'afterRate'. Why you need add second emit ? :)

Comment (ouraios) [reply Rate.vue]: Because i need to bind rate value to the v-model i provide in props.
Because i use v-for to loop through items and each time can have multiple rates items/value and when rate value change it's hard to know to which item this rate value is related too.

Comment (SinanMtl) [reply Rate.vue]: Your code that added 'input' named emit same works with 'afterRate'. As I understand it this code will be helpful.

```html
<Rate v-for="(item, index) in ratelist" :key="index" :length="5" :value="item.value" @afterRate="afterRate($event, item)"></Rate>
```

```javascript
  import Rate from '@/components/Rate'

  export default {
    name: 'app',
    data () {
      return {
        ratelist: [
          {value: '1'},
          {value: '3'},
          {value: '2'}
        ]
      }
    },
    methods: {
      afterRate (val, item) {
        item.value = val
      }
    },
    components: {
      Rate
    }
  }
```

Comment (ouraios) [reply Rate.vue]: Here's a complete example : 

https://codepen.io/Ouraios/pen/prYQMV

As you can see i want to trigger an "ajax" updating my movies rating, currently i need to create a specific method just to handle your component value changing.

If i had a v-model binding with the input emit i wouldnt need this specific method at all, the afterRate event would just tell me when to update my movies through my ajax method directly.

To summarize it would be more optimized with the v-model binding.

Oh and et becareful with case sensitive event because HTML handle it pretty bad : https://github.com/vuejs/vue/issues/2308

If you accept my PR maybe then you could just had one event called "change" just to notify when a rating change.

Comment (SinanMtl) [reply Rate.vue]: Hımm, I understood what you mean. Thanks :)

**Comment by ouraios** [comment]
Thanks you for merging it !
